### PR TITLE
Only 1 spell may be cast by a single wizard each turn 

### PR DESCRIPTION
**Pull request by gullskatten**
(forfeit spells does not count).

### CODE REVIEW

**Comment by gullskatten** [comment]
This fixes 2 known bugs:

* Damage reduction spells are now only activated the turn AFTER it was cast.
* Spells could possibly be sent twice from the client device (which would cause players to receive "double" damage).

File: core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt [hunk 1]
```diff
@@ -15,35 +15,41 @@ class SpellExecutor {
...
-        }
-
-        if (spell.spell.spellMitigation > 0) {
-            mitigationSpells.add(
-                DamageReduction(
-                    spell.caster,
-                    turn + 1, // Damage reduction spells should be activated 1 turn after current
-                    spell.spell.duration,
-                    spell.spell.spellMitigation
+        if(isSpellValidForTurn(spell, turn)) {
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt:18]: 1: This is the only real change,  implemented a function to check if the spell was cast by a wizard already.

File: core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt [hunk 2]
```diff
@@ -15,35 +15,43 @@ class SpellExecutor {
...
-                DamageReduction(
-                    spell.caster,
-                    turn + 1, // Damage reduction spells should be activated 1 turn after current
-                    spell.spell.duration,
-                    spell.spell.spellMitigation
+            if (spell.spell.spellMitigation > 0) {
+                mitigationSpells.add(
+                    DamageReduction(
+                        spell.caster,
+                        turn,
```
Comment (gullskatten) [inline core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt:33]: Previously turn was incremented by 1 here - but this had already been accounted for in DamageReduction (leading to damage reduction being activated 1 turn in advance)

**Review by cherrybeans — APPROVED** [review]
Nice 💯  It's all coming together

**Pull request merged**